services/select.js

```diff
@@ -208,193 +167,250 @@ function addIframeOverlays(el) {
   }
 }
 
 /**
- * Scroll user to the component. "Weeee!" or "What the?"
- * @param {Element} el
+ * determine if a component has settings
+ * @param {object} options
+ * @returns {boolean}
  */
-function scrollToComponent(el) {
-  var toolBarHeight = 70,
-    componentBarHeight = 30,
-    pos = window.scrollY + el.getBoundingClientRect().top - toolBarHeight - componentBarHeight;
-
-  scrollToY(pos, 1500, 'easeInOutQuint');
+function hasSettings(options) {
+  return groups.getSettingsFields(options.data).length > 0;
 }
 
 /**
- * Add the parent's label to the component bar.
- * @param {Element} componentBar
- * @param {Element} parentEl
+ * unhide settings button and add handler
+ * @param {Element} selector
+ * @param {object} options
  */
-function addParentLabel(componentBar, parentEl) {
-  var ref = parentEl.getAttribute(references.referenceAttribute),
-    parentName = references.getComponentNameFromReference(ref),
-    el = dom.create(`<span class="label parent" title="Go to parent component: ${label(parentName)}">
-      <img src="${site.get('assetPath')}/media/components/clay-kiln/component-bar-parent.svg" alt="Go to Parent">
-      <span>${label(parentName)}</span>
-    </span>`);
-
-  el.addEventListener('click', function (e) {
-    e.stopPropagation();
-    // Select the parent.
-    return focus.unfocus().then(function () {
-      unselect();
-      select(parentEl);
-      scrollToComponent(parentEl);
-    }).catch(_.noop);
-  });
-  componentBar.appendChild(el);
+function addSettingsHandler(selector, options) {
+  var button = dom.find(selector, '.selected-action-settings');
+
+  if (hasSettings(options)) {
+    button.classList.remove(hidden);
+    button.addEventListener('click', function (e) {
+      e.stopPropagation();
+      // Open the settings overlay.
+      return focus.unfocus().then(function () {
+        return forms.open(options.ref, document.body);
+      }).catch(_.noop);
+    });
+  }
 }
 
 /**
- * Add drag within a component list.
- * @param {Element} componentBar
- * @param {Element} el (component element)
+ * add component list placeholder when the last component is deleted from a list
+ * @param {object} parent
  */
-function addDragOption(componentBar, el) {
-  // `drag` class is applied to the `img` and and selector elements to simplify dragula logic.
-  var selectedLabel = componentBar.querySelector('.selected-label'),
-    dragIcon = dom.create(`<img src="${site.get('assetPath')}/media/components/clay-kiln/component-bar-drag.svg" alt="Drag" class="drag-icon"></span>`);
-
-  selectedLabel.setAttribute('title', `Drag to reorder: ${label(selectedLabel.getAttribute('title'))}`);
-  el.classList.add('drag');
-  selectedLabel.insertBefore(dragIcon, selectedLabel.firstChild);
+function addListPlaceholder(parent) {
+  var listDiv = dom.find(parent.listEl, '.component-list-inner'),
+    children = _.filter(listDiv.children, function (child) {
+      return !_.includes(['SCRIPT', 'STYLE'], child.tagName);
```

Comment:
note: if a component is added, and its list element is its root element (e.g. `<aside data-uri="source-links" data-editable="content"></aside>`), any scripts or styles will end up wrapped inside the div.

idea: update the wrapper in the `component-list` decorator to _not_ wrap those